builtin/providers/aws/resource_aws_s3_bucket.go

```diff
@@ -183,6 +183,116 @@ func resourceAwsS3Bucket() *schema.Resource {
 				},
 			},
 
+			"lifecycle_rule": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+							ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
```

Comment:
is it also worth bringing this out to Validators.go yet the same as the other validators referenced in this part of the schema?